### PR TITLE
Fix duplicate release sections in changelog

### PR DESCRIPTION
**Pull request by arthexis**
## Summary
- deduplicate changelog generation logic to merge repeated release sections and entries
- add regression tests covering duplicate release commits and previous changelog merges
- regenerate CHANGELOG.rst to collapse duplicate release headings

## Testing
- pytest tests/test_changelog_builder.py

------
https://chatgpt.com/codex/tasks/task_e_68e04ac10cf08326b7c8f057672b6320

### CODE REVIEW

File: core/changelog.py [hunk 1]
```diff
@@ -146,13 +156,50 @@ def _merge_sections(
...
+                    title=old.title,
+                    entries=list(old.entries),
+                    version=None,
+                    date=None,
+                )
+                merged.insert(0, unreleased_section)
+            else:
+                for entry in old.entries:
+                    if entry not in unreleased_section.entries:
+                        unreleased_section.entries.append(entry)
```
[the remark this answers is not in the document]
Comment (arthexis) [reply core/changelog.py:181]: @codex fix it
